src/lib/expression/expression_precedence.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <stdint.h>
+
+namespace opossum {
+
+// Precedence levels for parenthesizing expression arguments. See AbstractExpression::_enclose_argument_as_column_name()
+enum class ExpressionPrecedence : uint32_t {
+  Highest = 0,
+  UnaryPredicate,
+  MultiplicationDivision,
+  AdditionSubtraction,
+  BinaryTernaryPredicate,
+  Logical
```

Comment:
Will look into this in more detail again. Ultimately, this is an matter for the sql-parser to handle correctly, but I'll write some precedence test for hyrise anyway, just because it's easy.